heareval/predictions/task_predictions.py

```diff
@@ -36,30 +49,65 @@ def __init__(self, nfeatures: int, nlabels: int, prediction_type: str):
         torch.nn.init.normal_(self.projection.weight)
         if prediction_type == "multilabel":
             self.activation: torch.nn.Module = torch.nn.Sigmoid()
+            self.logitloss = torch.nn.BCEWithLogitsLoss()
         elif prediction_type == "multiclass":
             self.activation = torch.nn.Softmax()
+            self.logit_loss = OneHotToCrossEntropyLoss()
         else:
             raise ValueError(f"Unknown prediction_type {prediction_type}")
 
-    def forward(self, x: torch.Tensor):
-        x, y = x
-        x = self.projection(x)
+    def forward_logit(self, x: torch.Tensor) -> torch.Tensor:
+        return self.projection(x)
+        return x
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.forward_logit(x)
         x = self.activation(x)
         return x
 
 
+class PredictionModel(pl.LightningModule):
+    def __init__(self, nfeatures: int, nlabels: int, prediction_type: str):
+        super().__init__()
+
+        self.predictor = RandomProjectionPrediction(nfeatures, nlabels, prediction_type)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.predictor(x)
```

Comment:
So is the activation function only applied during testing?

Reply:
yes, but it is in forward too